### PR TITLE
spec: add start of gov, lending and staking

### PR DESCRIPTION
**Pull request by marbar3778**
## Description

- add gov, lending and staking sections
- outstanding
-  [ ] type definitions & messages
-  [x] liquidation info
-  [ ] block rewards to lenders?

### CODE REVIEW

File: spec/staking.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# Staking
```
Comment (cmwaters) [inline spec/staking.md:1]: Is staking based per lending pool, or throughout the entire entropy protocol?

Comment (marbar3778) [reply spec/staking.md:1]: Right now staking will be separate from lending. We need to rewrite lending to use it as a staking pool.

File: spec/governance.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+The parameters of the protocol are:
+
+- C-ratio (Collateral Ratio)
+  - The collateral ratio of loans controls how much a user must put in for a loan. At the start it will be set to 175%, as the protocol matures it will be lowered.
+
+- Lending pools
+  - Lending pools represents which pools are available to lend and borrow from. Lending pools are created via governance.
+
+- Leverage
+  - Leverage is the amount of margin a trader can take. The higher leverage the protocol allows the more at risk the protocol itself becomes. If a trader takes a large amount of leverage in a illiquid market the protocol will have a harder time liquidating the position if the position does not work in favor of the trader
```
Comment (cmwaters) [inline spec/governance.md:18]: Can you tell me how this is different to the collateral ratio?

File: spec/lending.md [hunk 3]
```diff
@@ -0,0 +1,12 @@
+# Lending
+
+Lending is essential to a margin based protocol. In order to trade margin a user must be able to get the added funds required to over leverage themselves.
+There will be three users of the lending protocol.
+
+First, a user who wants to lend out their money. This user will come to the entropy protocol, add funds to a lending pool. The reason a user will want to do this is because there are users who will want to borrow money in order to margin trade.
+
+Second, a user who wants to borrow money. This user will create an account to send money to, this money will represent the collateral for the loan. The c-ratio will be 150%. This means for a loan of $100 the user must put up $150. A user will want to loan money for countless reasons, one being if they believe the underlying collateral token will raise in price.
```
Comment (cmwaters) [inline spec/lending.md:8]: Can collateral currency and the currency that is lent be different?

File: spec/governance.md (hunk 2 above)
Comment (marbar3778) [reply spec/governance.md:18]: collateral ratio is for a user who is lending funds while leverage is out of the scope of c-ratio. The protocol can decide to offer 5x margin thus in the scenario of trading 1 of x they only need to put up .2 of x. While a borrower would need to put up 1.5 of x

Comment (cmwaters) [reply spec/governance.md:18]: Okay so we have a multi purpose lending pool with different borrowing requirements. We will need to make sure that someone can't arbitrate over these different requirements i.e. pretend to do a margin trade when really only wanting to loan currency.

Although I agree principally with having just one lending pool to serve all the various requirements, we might want to consider implementing just for margin trading loans first and we can look to broaden the usage of the lending pool later

File: spec/lending.md (hunk 3 above)
Comment (cmwaters) [reply spec/lending.md:8]: What happens to this locked up collateral btw? Are we going to try to utilize it for something

File: spec/governance.md (hunk 2 above)
Comment (marbar3778) [reply spec/governance.md:18]: > Okay so we have a multi purpose lending pool with different borrowing requirements. We will need to make sure that someone can't arbitrate over these different requirements i.e. pretend to do a margin trade when really only wanting to loan currency.

Ill make a diagram for the flow. A margin trader should not be able to take a loan without doing a trade, it should be a single tx to take the loan and make the trade. Then selling of the position would return the loan.

File: spec/lending.md (hunk 3 above)
Comment (marbar3778) [reply spec/lending.md:8]: I don't think we can use it for anything. It has to be available for liquidation if it goes under the c-ratio.

**Comment by marbar3778** [comment]
leaving gov unfinished, when we get to that part we can extend it. 

The margin account has some notes on cross program interactions. The messages for trade, deleverage and liquidation needs written, this can be done once the contract is written.

**Pull request merged**